src/palace/manager/api/util/patron.py

```diff
@@ -69,27 +70,21 @@ def assert_borrowing_privileges(cls, patron):
         :raises OutstandingFines: If the patron has too many outstanding fines.
 
         """
+        if patron.block_reason == PatronData.NO_VALUE:
+            return
+
```

Comment:
It's not obvious to me why it is okay to move this ahead of checking whether the patron's card has expired. It seems unlikely that a patron with an expired card would have borrowing privileges.

If this case is actually allowed for some reason, it would be good to both (1) add a test with this scenario, and (2) update the docstring with some explanatory text.